### PR TITLE
Update release notes for 9.6.0

### PR DESCRIPTION
**Pull request by wjrosa**
Reference PR https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4418

## Changes proposed in this Pull Request:

This PR updates the release notes for 9.6.0, making some minor edits to the new version details (such as separating changes by type and importance).

## Testing instructions

Visual inspection should be sufficient, but you can also preview the content by copying the contents of readme.txt and previewing them at https://wpreadme.com/.

---

-   [x] Covered with tests (or have a good reason not to test in description ☝️)
-   [x] Tested on mobile (or does not apply)

### Changelog entry

-   [x] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

This PR only touches the `readme.txt` file.

</details>

**Post merge**

-   [ ] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)
-   [ ] Added the [needs docs label](https://github.com/woocommerce/woocommerce-gateway-stripe/labels?q=docs) (or does not apply)
-   [ ] Included this PR in the Release Thread scope (or does not apply)

### CODE REVIEW

File: readme.txt [hunk 1]
```diff
@@ -112,62 +112,56 @@ If you get stuck, you can ask for help in the [Plugin Forum](https://wordpress.o
...
+* Update - Add prefix to the custom database cache keys
+* Update - Support block checkout custom fields when using express payment methods like Apple Pay and Google Pay
+* Update - Remove Payment Method Configurations fallback cache
+* Update - Update deprecation notice message to specify that legacy checkout experience has been deprecated since version 9.6.0
+* Update - Remove legacy checkout checkbox from settings
 * Fix - Correctly notifies customers and merchants of a failed refund and reverts the refunded status
-* Update - Legacy checkout experience has been deprecated, new checkout experience is now the default for all sites
+* Fix - Sends missing information to Stripe when completing transactions with WeChat Pay, Blik and Klarna, using the Optimized Checkout.
+* Fix - Void intent when cancelling an uncaptured order
+* Fix - Apply shipping country restrictions to Express Checkout
```
Comment (daledupreez) [inline readme.txt]: I vote that we order these entries based on their likely importance to users. So I would recommend something more like the following, which shifts a few items to the internal list.
```suggestion
* Fix - Apply shipping country restrictions to Express Checkout
* Update - Support block checkout custom fields when using express payment methods like Apple Pay and Google Pay
* Fix - Correctly notifies customers and merchants of a failed refund and reverts the refunded status
* Fix - Void intent when cancelling an uncaptured order
* Update - Remove BACS from the unsupported ‘change payment method for subscription’ page
* Update - Remove verification steps for Apple Pay domain registration, as this is no longer required by Stripe
* Update - Update deprecation notice message to specify that legacy checkout experience has been deprecated since version 9.6.0

**Internal Changes and Upcoming Features**

* Fix - Sends missing information to Stripe when completing transactions with WeChat Pay, Blik and Klarna, using the Optimized Checkout
* Update - Remove legacy checkout checkbox from settings
* Update - Remove Payment Method Configurations fallback cache
* Update - Add prefix to the custom database cache keys
```

File: readme.txt [hunk 2]
```diff
@@ -112,62 +112,56 @@ If you get stuck, you can ask for help in the [Plugin Forum](https://wordpress.o
...
-* Update - Removes the customization of individual payment method titles and descriptions
-* Fix - Fixes some inconsistencies related to the Optimized Checkout feature and improves its unit tests
-* Add - Introduces a new marketing note to promote BNPLs (Buy Now Pay Later) payment methods (Klarna and Affirm) on WooCommerce admin home page
-* Fix - Throws a specific exception on an edge case where a saved payment method could not be found when processing an order in the new checkout experience
-* Fix - Checks if the store has other BNPL extensions installed before displaying the promotional banner
+**New Features**
+
+* Legacy checkout experience has been deprecated, new checkout experience is now the default for all sites
+* Voucher payment methods (Boleto, Multibanco, and Oxxo) can now be used when purchasing subscriptions if manual renewals are enabled or required
+* Adds a new filter (wc_stripe_is_optimized_checkout_available) to allow merchants to test the Optimized Checkout feature earlier
```
Comment (daledupreez) [inline readme.txt]: Should this entry be moved to _Internal Changes and Upcoming Features_?

File: readme.txt [hunk 3]
```diff
@@ -112,62 +112,56 @@ If you get stuck, you can ask for help in the [Plugin Forum](https://wordpress.o
...
+* Fix - Void intent when cancelling an uncaptured order
+* Fix - Apply shipping country restrictions to Express Checkout
+
+**Other Fixes**
+
 * Fix - Fixes an edge case where the express payment method buttons would not be displayed on the checkout if taxes used to be enabled
-* Add - Voucher payment methods (Boleto, Multibanco, and Oxxo) can now be used when purchasing subscriptions if manual renewals are enabled or required
-* Add - Adds a new filter (`wc_stripe_is_optimized_checkout_available`) to allow merchants to test the Optimized Checkout feature earlier
-* Fix - Restricts the BNPLs promotional banner to only be displayed after version 9.7.0
+* Fix - Fixes a fatal error when the fingerprint property is not available for a card payment method
```
Comment (daledupreez) [inline readme.txt:144]: Did we miss including this fix in 9.5.3?

File: readme.txt (hunk 1 above)
Comment (wjrosa) [reply readme.txt]: Agreed! Applied it in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4460/commits/03870ff4664f52c89b04a8d906241f10c2a1333d. I tried to auto apply here, but got this error:
Screenshot 2025-07-03 at 18 40 43

File: readme.txt (hunk 2 above)
Comment (wjrosa) [reply readme.txt]: I think it makes sense in any of the sections, but I moved it anyway in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4460/commits/0992cee378bc6e446ea94e058ab95631fb8df75b

File: readme.txt (hunk 3 above)
Comment (wjrosa) [reply readme.txt:144]: Yes 👍 . I don't see it in the previous changelog entries or in the [9.5.3 release description](https://github.com/woocommerce/woocommerce-gateway-stripe/releases/tag/9.5.3). I also see it was added to 9.6.0 milestone in the original PR https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4339

File: readme.txt [hunk 4]
```diff
@@ -112,62 +112,56 @@ If you get stuck, you can ask for help in the [Plugin Forum](https://wordpress.o
...
+* Fix - Hide future payments message from payment element when manual renewal is required
+* Fix - Fix a rare warning when searching customers with missing name
+* Fix - Fix legacy deprecation notice displayed on new plugin installs
+* Fix - When the user is deleted via WP CLI, take into account the environment type before detaching their payment methods
+* Fix - Throws a specific exception on an edge case where a saved payment method could not be found when processing an order in the new checkout experience
+* Fix - Register Express Checkout script before use to restore buttons on “order-pay” pages
+
+**Internal Changes and Upcoming Features**
+
+* Update - Remove legacy checkout checkbox from settings
```
Comment (Mayisha) [inline readme.txt]: I think this one should go in `**Important Fixes and Updates**`. As we are removing this from the UI, this is an important change for the legacy users.

**Review by Mayisha — COMMENTED** [review]
I recommended one entry to move to a different section. Other than that, it looks good to me 👍

File: readme.txt (hunk 4 above)
Comment (wjrosa) [reply readme.txt]: That's a good point. Moved it in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4460/commits/9d55c94ddee77364f5118b51c2d9d1786d628615 (also sorted some other entries by type)

**Review by Mayisha — APPROVED** [review]

File: readme.txt [hunk 5]
```diff
@@ -112,62 +112,56 @@ If you get stuck, you can ask for help in the [Plugin Forum](https://wordpress.o
...
+* Show an icon beside the payment methods that support automatic recurring payments
+
+**Important Fixes and Updates**
+
+* Update - Support block checkout custom fields when using express payment methods like Apple Pay and Google Pay
+* Update - Remove legacy checkout checkbox from settings
+* Update - Remove BACS from the unsupported ‘change payment method for subscription’ page
+* Update - Remove verification steps for Apple Pay domain registration, as this is no longer required by Stripe
+* Update - Update deprecation notice message to specify that legacy checkout experience has been deprecated since version 9.6.0
+* Fix - Apply shipping country restrictions to Express Checkout
```
Comment (daledupreez) [inline readme.txt]: I think the express checkout shipping restrictions should be the second entry, as that is a pretty critical change that impacts inbound payments.
```suggestion
* Update - Support block checkout custom fields when using express payment methods like Apple Pay and Google Pay
* Fix - Apply shipping country restrictions to Express Checkout
* Update - Remove legacy checkout checkbox from settings
* Update - Remove BACS from the unsupported ‘change payment method for subscription’ page
* Update - Remove verification steps for Apple Pay domain registration, as this is no longer required by Stripe
* Update - Update deprecation notice message to specify that legacy checkout experience has been deprecated since version 9.6.0
```

File: readme.txt [hunk 6]
```diff
@@ -112,62 +112,56 @@ If you get stuck, you can ask for help in the [Plugin Forum](https://wordpress.o
 
 = 9.6.0 - xxxx-xx-xx =
 
-* Update - Removes the change display order feature from the settings page when the Optimized Checkout is enabled
-* Update - Removes the customization of individual payment method titles and descriptions
-* Fix - Fixes some inconsistencies related to the Optimized Checkout feature and improves its unit tests
-* Add - Introduces a new marketing note to promote BNPLs (Buy Now Pay Later) payment methods (Klarna and Affirm) on WooCommerce admin home page
```
Comment (daledupreez) [inline readme.txt]: These four items are all missing from the release notes. Based on other optimized checkout fixes being included, it's not clear if that was intentional.

File: readme.txt [hunk 7]
```diff
@@ -112,62 +112,56 @@ If you get stuck, you can ask for help in the [Plugin Forum](https://wordpress.o
...
-* Fix - Throws a specific exception on an edge case where a saved payment method could not be found when processing an order in the new checkout experience
-* Fix - Checks if the store has other BNPL extensions installed before displaying the promotional banner
```
Comment (daledupreez) [inline readme.txt:120]: Another missing entry

File: readme.txt [hunk 8]
```diff
@@ -112,62 +112,56 @@ If you get stuck, you can ask for help in the [Plugin Forum](https://wordpress.o
...
 * Fix - Correctly notifies customers and merchants of a failed refund and reverts the refunded status
-* Update - Legacy checkout experience has been deprecated, new checkout experience is now the default for all sites
+* Fix - Void intent when cancelling an uncaptured order
+
+**Other Fixes**
+
 * Fix - Fixes an edge case where the express payment method buttons would not be displayed on the checkout if taxes used to be enabled
-* Add - Voucher payment methods (Boleto, Multibanco, and Oxxo) can now be used when purchasing subscriptions if manual renewals are enabled or required
-* Add - Adds a new filter (`wc_stripe_is_optimized_checkout_available`) to allow merchants to test the Optimized Checkout feature earlier
-* Fix - Restricts the BNPLs promotional banner to only be displayed after version 9.7.0
```
Comment (daledupreez) [inline readme.txt:126]: Another missing entry. Maybe an internal change? Though it may also be intentional that it should fold into the entry for the new promotional banner. Maybe we should mention the intent for it to be shown in 9.7.0 in the main entry.

File: readme.txt [hunk 9]
```diff
@@ -112,62 +112,56 @@ If you get stuck, you can ask for help in the [Plugin Forum](https://wordpress.o
...
-* Dev - Implements the PSR-4 autoloading standard for the plugin unit tests (PHP).
-* Fix - Sends missing information to Stripe when completing transactions with WeChat Pay, Blik and Klarna, using the Optimized Checkout.
-* Dev - Moves the main Stripe class to a new file.
-* Dev - Renames all PHP Unit test files to follow the PSR-4.
+* Add - [OC] Adds a new filter (wc_stripe_is_optimized_checkout_available) to allow merchants to test the Optimized Checkout feature earlier
+* Fix - [OC] Sends missing information to Stripe when completing transactions with WeChat Pay, Blik and Klarna, using the Optimized Checkout
+* Fix - [OC] Fixes the availability of the saving payment method checkbox in the classic checkout when the Optimized Checkout is enabled and signup is disabled during checkout
+* Fix - [OC] Makes payment methods dynamically available on the shortcode checkout when the Optimized Checkout is enabled depending on the saving method checkbox value
+* Fix - [OC] Fixes the payment method title when using the classic checkout with the Optimized Checkout enabled
+* Dev - Deprecates the WC_Stripe_Order class and removes its inclusion call
```
Comment (daledupreez) [inline readme.txt]: This change has compatibility implications for users who are referencing or using the class. It may be worth flagging this in the important updates section.

File: readme.txt [hunk 10]
```diff
@@ -112,62 +112,56 @@ If you get stuck, you can ask for help in the [Plugin Forum](https://wordpress.o
...
-* Dev - Implements the PSR-4 autoloading standard for the plugin unit tests (PHP).
-* Fix - Sends missing information to Stripe when completing transactions with WeChat Pay, Blik and Klarna, using the Optimized Checkout.
-* Dev - Moves the main Stripe class to a new file.
-* Dev - Renames all PHP Unit test files to follow the PSR-4.
+* Add - [OC] Adds a new filter (wc_stripe_is_optimized_checkout_available) to allow merchants to test the Optimized Checkout feature earlier
+* Fix - [OC] Sends missing information to Stripe when completing transactions with WeChat Pay, Blik and Klarna, using the Optimized Checkout
+* Fix - [OC] Fixes the availability of the saving payment method checkbox in the classic checkout when the Optimized Checkout is enabled and signup is disabled during checkout
+* Fix - [OC] Makes payment methods dynamically available on the shortcode checkout when the Optimized Checkout is enabled depending on the saving method checkbox value
+* Fix - [OC] Fixes the payment method title when using the classic checkout with the Optimized Checkout enabled
```
Comment (daledupreez) [inline readme.txt]: Rather than using `[OC]`, I think we should use `[Optimized Checkout]`, as `OC` is pretty meaningless to anyone outside of the development team and a handful of others.
```suggestion
* Add - [Optimized Checkout] Adds a new filter (wc_stripe_is_optimized_checkout_available) to allow merchants to test the Optimized Checkout feature earlier
* Fix - [Optimized Checkout] Sends missing information to Stripe when completing transactions with WeChat Pay, Blik and Klarna, using the Optimized Checkout
* Fix - [Optimized Checkout] Fixes the availability of the saving payment method checkbox in the classic checkout when the Optimized Checkout is enabled and signup is disabled during checkout
* Fix - [Optimized Checkout] Makes payment methods dynamically available on the shortcode checkout when the Optimized Checkout is enabled depending on the saving method checkbox value
* Fix - [Optimized Checkout] Fixes the payment method title when using the classic checkout with the Optimized Checkout enabled
```

File: readme.txt [hunk 11]
```diff
@@ -112,62 +112,56 @@ If you get stuck, you can ask for help in the [Plugin Forum](https://wordpress.o
...
 * Fix - Fixes an edge case where the express payment method buttons would not be displayed on the checkout if taxes used to be enabled
-* Add - Voucher payment methods (Boleto, Multibanco, and Oxxo) can now be used when purchasing subscriptions if manual renewals are enabled or required
-* Add - Adds a new filter (`wc_stripe_is_optimized_checkout_available`) to allow merchants to test the Optimized Checkout feature earlier
-* Fix - Restricts the BNPLs promotional banner to only be displayed after version 9.7.0
+* Fix - Fixes a fatal error when the fingerprint property is not available for a card payment method
+* Fix - Fix payment method title display when new payment settings experience is enabled
+* Fix - Prevent styles from non-checkout pages affecting the appearance of Stripe element
+* Fix - Send correct attribute when setting the default payment method
+* Fix - Hide future payments message from payment element when manual renewal is required
+* Fix - Fix a rare warning when searching customers with missing name
```
Comment (daledupreez) [inline readme.txt:149]: Good call on changing this to a `Fix`.

File: readme.txt [hunk 12]
```diff
@@ -112,62 +112,56 @@ If you get stuck, you can ask for help in the [Plugin Forum](https://wordpress.o
...
+* Fix - Fix legacy deprecation notice displayed on new plugin installs
+* Fix - When the user is deleted via WP CLI, take into account the environment type before detaching their payment methods
+* Fix - Throws a specific exception on an edge case where a saved payment method could not be found when processing an order in the new checkout experience
+* Fix - Register Express Checkout script before use to restore buttons on “order-pay” pages
+
+**Internal Changes and Upcoming Features**
+
+* Update - Remove Payment Method Configurations fallback cache
+* Update - Add prefix to the custom database cache keys
```
Comment (daledupreez) [inline readme.txt:159]: This feels like it should be classified as a `Dev` entry rather than an `Update`, but I don't think this is blocking either way.

File: readme.txt [hunk 13]
```diff
@@ -112,62 +112,56 @@ If you get stuck, you can ask for help in the [Plugin Forum](https://wordpress.o
...
+* Fix - Hide future payments message from payment element when manual renewal is required
+* Fix - Fix a rare warning when searching customers with missing name
+* Fix - Fix legacy deprecation notice displayed on new plugin installs
+* Fix - When the user is deleted via WP CLI, take into account the environment type before detaching their payment methods
+* Fix - Throws a specific exception on an edge case where a saved payment method could not be found when processing an order in the new checkout experience
+* Fix - Register Express Checkout script before use to restore buttons on “order-pay” pages
+
+**Internal Changes and Upcoming Features**
+
+* Update - Remove Payment Method Configurations fallback cache
```
Comment (daledupreez) [inline readme.txt:158]: Not sure this should be shifted to an Update rather than a Tweak, as this is mostly an implementation detail.

File: readme.txt [hunk 14]
```diff
@@ -112,62 +112,56 @@ If you get stuck, you can ask for help in the [Plugin Forum](https://wordpress.o
...
-* Fix - Prevent styles from non-checkout pages affecting the appearance of Stripe element.
 * Dev - Add e2e tests for BLIK
 * Dev - Add e2e tests for BECS
-* Fix - Send correct attribute when setting the default payment method.
-* Dev - Build dynamic WordPress and WooCommerce dependencies for unit tests.
-* Fix - Void intent when cancelling an uncaptured order
-* Fix - Hide future payments message from payment element when manual renewal is required
-* Tweak - Fix a rare warning when searching customers with missing name
-* Fix - Apply shipping country restrictions to Express Checkout
+* Dev - Build dynamic WordPress and WooCommerce dependencies for unit tests
```
Comment (daledupreez) [inline readme.txt:180]: Klarna was missing from the updated readme/changelog.
```suggestion
* Dev - Add Klarna e2e tests
* Dev - Build dynamic WordPress and WooCommerce dependencies for unit tests
```

File: readme.txt [hunk 15]
```diff
@@ -112,62 +112,56 @@ If you get stuck, you can ask for help in the [Plugin Forum](https://wordpress.o
...
-* Update - Remove verification steps for Apple Pay domain registration, as this is no longer required by Stripe
-* Fix - When the user is deleted via WP CLI, take into account the environment type before detaching their payment methods
-* Tweak - Add prefix to the custom database cache keys
-* Update - Support block checkout custom fields when using express payment methods like Apple Pay and Google Pay
 * Dev - Fix failing optimized checkout e2e test due to incorrect order of operations
-* Tweak - Remove Payment Method Configurations fallback cache
-* Dev - Add Klarna e2e tests
-* Tweak - Update deprecation notice message to specify that legacy checkout experience has been deprecated since version 9.6.0
-* Update - Remove legacy checkout checkbox from settings
-* Add - Introduced `wc_stripe_force_save_payment_method` filter
```
Comment (daledupreez) [inline readme.txt:161]: This entry is missing from the readme -- we should add it, either in the other important updates section.

File: readme.txt [hunk 16]
```diff
@@ -112,62 +112,56 @@ If you get stuck, you can ask for help in the [Plugin Forum](https://wordpress.o
...
-* Update - Express Checkout: introduce new WP actions for supporting custom checkout fields for classic, shortcode-based checkout
```
Comment (daledupreez) [inline readme.txt:162]: This entry is also missing from the new readme. If we're folding this into the custom checkout fields entry, we need to mention that new actions are involved and specify that it is for block and classic shortcode checkout - the current entry only mentions block checkout.

File: readme.txt [hunk 17]
```diff
@@ -112,62 +112,56 @@ If you get stuck, you can ask for help in the [Plugin Forum](https://wordpress.o
...
-* Fix - Fixes page crash when Klarna payment method is not supported in the merchant's country by returning an empty array instead of throwing an error
```
Comment (daledupreez) [inline readme.txt:163]: This fix is missing.

File: readme.txt [hunk 18]
```diff
@@ -112,62 +112,56 @@ If you get stuck, you can ask for help in the [Plugin Forum](https://wordpress.o
...
-* Fix - Fix payment processing for $0 subscription with recurring coupon
-* Dev - Add e2e tests to cover Affirm purchase flow
-* Dev - Improve e2e tests of some of the LPMs
-* Fix - Add safety check when checking error object
-* Fix - Correctly handle countries without states when using the express payment methods
```
Comment (daledupreez) [inline readme.txt]: All of these entries are missing from the new changelog, with the last one being pretty important.

File: readme.txt [hunk 19]
```diff
@@ -112,62 +112,56 @@ If you get stuck, you can ask for help in the [Plugin Forum](https://wordpress.o
...
-* Update - Include extension data from block checkout when submitting an express checkout order
```
Comment (daledupreez) [inline readme.txt:169]: This entry is missing from the readme.

File: readme.txt [hunk 20]
```diff
@@ -112,62 +112,56 @@ If you get stuck, you can ask for help in the [Plugin Forum](https://wordpress.o
...
-* Fix - Add order locking when processing payment redirects, to mitigate cases of double status updates
```
Comment (daledupreez) [inline readme.txt:170]: This entry is also missing from the updated readme.

**Review by daledupreez — CHANGES_REQUESTED** [review]
I have lots of comments, and noticed a number of entries that got dropped.

File: readme.txt [hunk 21]
```diff
@@ -112,62 +112,56 @@ If you get stuck, you can ask for help in the [Plugin Forum](https://wordpress.o
...
+* Dev - Build dynamic WordPress and WooCommerce dependencies for unit tests
 * Dev - Prevent changelog entries with trailing periods
-* Fix - Fix legacy deprecation notice displayed on new plugin installs
-* Update - Remove verification steps for Apple Pay domain registration, as this is no longer required by Stripe
-* Fix - When the user is deleted via WP CLI, take into account the environment type before detaching their payment methods
-* Tweak - Add prefix to the custom database cache keys
-* Update - Support block checkout custom fields when using express payment methods like Apple Pay and Google Pay
 * Dev - Fix failing optimized checkout e2e test due to incorrect order of operations
-* Tweak - Remove Payment Method Configurations fallback cache
-* Dev - Add Klarna e2e tests
```
Comment (daledupreez) [inline readme.txt:158]: This Klarna entry was also missing.

File: readme.txt (hunk 7 above)
Comment (daledupreez) [reply readme.txt:120]: Added in 072172e71f8260fc5e464b64434372598428fc31

File: readme.txt (hunk 8 above)
Comment (daledupreez) [reply readme.txt:126]: Added in 072172e71f8260fc5e464b64434372598428fc31

File: readme.txt (hunk 21 above)
Comment (daledupreez) [reply readme.txt:158]: Added in 072172e71f8260fc5e464b64434372598428fc31

File: readme.txt (hunk 15 above)
Comment (daledupreez) [reply readme.txt:161]: Added in 072172e71f8260fc5e464b64434372598428fc31

File: readme.txt (hunk 16 above)
Comment (daledupreez) [reply readme.txt:162]: Added in 072172e71f8260fc5e464b64434372598428fc31

File: readme.txt (hunk 17 above)
Comment (daledupreez) [reply readme.txt:163]: Added in 072172e71f8260fc5e464b64434372598428fc31

File: readme.txt (hunk 19 above)
Comment (daledupreez) [reply readme.txt:169]: Added in 599133efb4d45349653f9bf1c6742aa0de4ef38a

File: readme.txt (hunk 20 above)
Comment (daledupreez) [reply readme.txt:170]: Added in 599133efb4d45349653f9bf1c6742aa0de4ef38a

**Review by daledupreez — APPROVED** [review]
Approving for now, but I would really appreciate one more round of checks before we merge.

File: readme.txt (hunk 9 above)
Comment (wjrosa) [reply readme.txt]: Moved it in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4460/commits/8529f16e0aea5fa296329412d31b3906025c1e0b
